src/util/userHelpers.js

```diff
@@ -176,3 +176,18 @@ export const hasPermissionToInitiateTransactions = currentUser => {
  * @returns {Boolean} true if currentUser has been approved (state is 'active').
  */
 export const isUserAuthorized = currentUser => currentUser?.attributes?.state === 'active';
+
+/**
+ * Check if currentUser has permission to view listing and user data on a private marketplace.
+ *
+ * @param {Object} currentUser API entity
+ * @returns {Boolean} true if currentUser has permission to view listing and user data on a private marketplace.
+ */
+export const hasReadPermission = currentUser => {
+  if (currentUser?.id && !currentUser?.effectivePermissionSet?.id) {
+    console.warn(
+      '"effectivePermissionSet" relationship is not defined or included to the fetched currentUser entity.'
+    );
+  }
+  return currentUser?.effectivePermissionSet?.attributes?.read === 'permission/allow';
+};
```

Comment:
It might also make sense to add a link to API reference > currentUser's PermissionSet 
https://www.sharetribe.com/api-reference/marketplace.html#currentuser-permissionset

...but I'm not sure when it will get every permission mentioned there.

Reply:
Fixed, link added to hasPermission function comments